### PR TITLE
GH-39734: [Java] Bump org.codehaus.mojo:exec-maven-plugin from 1.6.0 to 3.1.1

### PR DESCRIPTION
**Pull request by vibhatha**
### Rationale for this change

This PR was created to replace https://github.com/apache/arrow/pull/39374 and do the necessary changes for `org.codehaus.mojo` upgrade to take place. 

### What changes are included in this PR?

The changes to the `org.codehaus.mojo` version and an upgrade on the maven version used in the `.env`. 

### Are these changes tested?

Tested locally, but this requires a CI verification on Java.

### Are there any user-facing changes?

No

### CODE REVIEW

**Comment by vibhatha** [comment]
@github-actions crossbow submit *java*

**Review by raulcd — CHANGES_REQUESTED** [review]
Thanks @vibhatha ! 
The integration failure seems related:
```
 Error:  Failed to execute goal org.codehaus.mojo:exec-maven-plugin:3.1.1:exec (run-java-benchmarks) on project arrow-performance: The plugin org.codehaus.mojo:exec-maven-plugin:3.1.1 requires Maven version 3.6.3 -> [Help 1]
Error:  
Error:  To see the full stack trace of the errors, re-run Maven with the -e switch.
Error:  Re-run Maven using the -X switch to enable full debug logging.
Error:  
Error:  For more information about the errors and possible solutions, please read the following articles:
Error:  [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginIncompatibleException
Error:  
Error:  After correcting the problems, you can resume the build with the command
Error:    mvn <goals> -rf :arrow-performance
```
Also, this should not be `MINOR` and we should create an issue for it.

**Comment by vibhatha** [comment]
@raulcd let me check this. 👍

**Comment by vibhatha** [comment]
@raulcd is there a way to re-run the CIs. 

I looked into the logs and noticed that the environment already has MAVEN 3.6.3 

```bash
Error:  Failed to execute goal org.codehaus.mojo:exec-maven-plugin:3.1.1:exec (run-java-benchmarks) on project arrow-performance: The plugin org.codehaus.mojo:exec-maven-plugin:3.1.1 requires Maven version 3.6.3 -> [Help 1]
Error:  
Error:  To see the full stack trace of the errors, re-run Maven with the -e switch.
Error:  Re-run Maven using the -X switch to enable full debug logging.
Error:  
Error:  For more information about the errors and possible solutions, please read the following articles:
Error:  [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginIncompatibleException
Error:  
Error:  After correcting the problems, you can resume the build with the command
Error:    mvn <goals> -rf :arrow-performance
1
Error: `docker-compose --file /home/runner/work/arrow/arrow/docker-compose.yml run --rm -e ARCHERY_DEFAULT_BRANCH=main -e ARCHERY_INTEGRATION_WITH_RUST=1 conda-integration` exited with a non-zero exit code 1, see the process log above.

The docker-compose command was invoked with the following parameters:

Defaults defined in .env:
  ALMALINUX: 8
  ALPINE_LINUX: 3.16
  ARCH: amd64
  ARCH_ALIAS: x86_64
  ARCH_SHORT: amd64
  ARROW_R_DEV: TRUE
  BUILDKIT_INLINE_CACHE: 1
  CLANG_TOOLS: 14
  COMPOSE_DOCKER_CLI_BUILD: 1
  CONAN: gcc10
  CUDA: 11.2.2
  DASK: latest
  DEBIAN: 11
  DEVTOOLSET_VERSION: 
  DOCKER_BUILDKIT: 1
  DOCKER_VOLUME_PREFIX: 
  DOTNET: 7.0
  FEDORA: 38
  GCC_VERSION: 
  GO: 1.19.13
  HDFS: 3.2.1
  JDK: 8
  KARTOTHEK: latest
  LLVM: 14
  MAVEN: 3.6.3
  NODE: 18
  NUMBA: latest
  NUMPY: latest
  PANDAS: latest
  PYTHON: 3.8
  PYTHON_WHEEL_WINDOWS_IMAGE_REVISION: 2023-08-02
  R: 4.2
  REPO: apache/arrow-dev
  R_CUSTOM_CCACHE: false
  R_IMAGE: ubuntu-gcc-release
  R_ORG: rhub
  R_PRUNE_DEPS: FALSE
  R_TAG: latest
  SPARK: master
  STATICCHECK: v0.4.5
  TURBODBC: latest
  TZ: UTC
  UBUNTU: 20.04
  ULIMIT_CORE: -1
```

**Comment by lidavidm** [comment]
I kicked the job.

**Comment by vibhatha** [comment]
Thanks @lidavidm

**Comment by lidavidm** [comment]
It appears it's still failing. You'll want to dig into the image, perhaps it's picking up the wrong Maven.

**Comment by vibhatha** [comment]
Yeah. Will analyze this. Thanks @lidavidm

**Comment by vibhatha** [comment]
@raulcd created a ticket: https://github.com/apache/arrow/issues/39734

How can we link to this PR? :slightly_smiling_face:

**Comment by vibhatha** [comment]
Thanks @assignUser

**Comment by vibhatha** [comment]
@github-actions crossbow submit *java*

File: docker-compose.yml [hunk 1]
```diff
@@ -1711,7 +1711,7 @@ services:
         jdk: 17
         # conda-forge doesn't have 3.5.4 so pinning explicitly, but this should
         # be set to ${MAVEN}
-        maven: 3.5
+        maven: 3.6.3
```
[the remark this answers is not in the document]
Comment (vibhatha) [reply docker-compose.yml]: Updated.

**Comment by vibhatha** [comment]
@github-actions crossbow submit *java*

**Review by lidavidm — APPROVED** [review]

**Comment by vibhatha** [comment]
@github-actions crossbow submit *java*

**Comment by lidavidm** [comment]
@raulcd any more comments here?

**Review by raulcd — APPROVED** [review]
Thanks!

**Pull request merged**